### PR TITLE
spectacle: add dependencies required to annotate screenshots

### LINKED ISSUE #129676: Plasma's Spectacle does not support annotating screenshots
**Describe the bug**
Starting with version 20.12, Spectacle has annotation support, utilizing [kImageAnnotator](https://github.com/ksnip/kImageAnnotator). Most big distributions already ship Spectacle with the feature enabled. I assume kImageAnnotator needs to be packaged and integrated with the existing Spectacle package.

**Screenshots**

Screenshot of the functionality from the official KDE docs:

Annotate

**Additional context**
- [Request in KDE's Bugzilla](https://bugs.kde.org/show_bug.cgi?id=372464)
- [The commit which added the feature](https://invent.kde.org/graphics/spectacle/-/commit/fa60f99470485143015b547b03ecba7811569480)

**Notify maintainers**

@ttuegel 

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

### PR DESCRIPTION
**Pull request by FliegendeWurst**
###### Motivation for this change
Closes #129676

###### Things done
- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files: annotation editor works
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 130007` at a3738776 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/a8bc5196dade80cb569f5013724eb2cd">libsForQt514.spectacle</a></li>
  </ul>
</details>
<details>
  <summary>1 package skipped due to time constraints:</summary>
  <ul>
    <li>libsForQt512.spectacle</li>
  </ul>
</details>
<details>
  <summary>7 packages built successfully:</summary>
  <ul>
    <li>libsForQt5.kcolorpicker (libsForQt515.kcolorpicker ,plasma5Packages.kcolorpicker)</li>
    <li>libsForQt5.kimageannotator (libsForQt515.kimageannotator ,plasma5Packages.kimageannotator)</li>
    <li>libsForQt5.spectacle (libsForQt515.spectacle ,plasma5Packages.spectacle)</li>
    <li>libsForQt512.kcolorpicker</li>
    <li>libsForQt512.kimageannotator</li>
    <li>libsForQt514.kcolorpicker</li>
    <li>libsForQt514.kimageannotator</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

File: pkgs/applications/kde/spectacle.nix [hunk 1]
```diff
@@ -14,7 +14,7 @@ mkDerivation {
   buildInputs = [
     kconfig kcoreaddons kdbusaddons kdeclarative ki18n kio knotifications
     kscreen kwidgetsaddons kwindowsystem kxmlgui libkipi qtx11extras xcb-util-cursor
-    knewstuff kwayland
+    knewstuff kwayland kcolorpicker kimageannotator
   ];
   postPatch = ''
     substituteInPlace desktop/org.kde.spectacle.desktop.cmake \
```
Comment (SuperSandro2000) [inline pkgs/applications/kde/spectacle.nix:20]: Please set meta

```nix
  meta = with lib; {
    broken = versionOlder qtbase.version "5.15";
  };
```

Comment (FliegendeWurst) [reply pkgs/applications/kde/spectacle.nix:23]: Thanks for the suggestion. I'm not sure why @r-rmcgibbo tried to build `libsForQt514.spectacle`, it was already marked as broken *without* this meta setting.

**Review by Artturin — APPROVED** [review]
Result of `nixpkgs-review pr 130007` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>libsForQt512.spectacle</li>
    <li>libsForQt514.spectacle</li>
  </ul>
</details>
<details>
  <summary>7 packages built:</summary>
  <ul>
    <li>libsForQt5.kcolorpicker (libsForQt515.kcolorpicker ,plasma5Packages.kcolorpicker)</li>
    <li>libsForQt5.kimageannotator (libsForQt515.kimageannotator ,plasma5Packages.kimageannotator)</li>
    <li>libsForQt5.spectacle (libsForQt515.spectacle ,plasma5Packages.spectacle)</li>
    <li>libsForQt512.kcolorpicker</li>
    <li>libsForQt512.kimageannotator</li>
    <li>libsForQt514.kcolorpicker</li>
    <li>libsForQt514.kimageannotator</li>
  </ul>
</details>

**Pull request merged**